Random Users/Networking/APIController.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  APIController.swift
+//  Random Users
+//
+//  Created by conner on 5/15/20.
+//  Copyright © 2020 Erica Sadun. All rights reserved.
+//
+
+import Foundation
+
+class APIController {
+    
+    enum NetworkError: Error {
+        case noData
+        case fetchError
+        case unknownError
+    }
+    
+    typealias CompletionHandler = (Result<Bool, NetworkError>) -> Void
+    
+    var users: [User] = []
+    let usersLock = NSLock()
+    let baseURL = URL(string: "https://randomuser.me/api/?format=json&inc=name,email,phone,picture&results=50")!
+    
+    func getUsers(completion: @escaping CompletionHandler = { _ in }) {
+        URLSession.shared.dataTask(with: baseURL) { data, _, error in
+            if let error = error {
+                NSLog("Failed fetch: \(error)")
+                return completion(.failure(.fetchError))
```

Comment:
It would be good to switch to the main queue here since the code calling this likely has that assumption.